crates/atiny-checker/src/lib.rs

```diff
@@ -64,7 +64,7 @@ fn unify_hole(
             if let MonoType::Hole(oth_ref) = &*other {
                 match oth_ref.get() {
                     Hole::Empty(lvl2) if lvl2 > lvl => {
-                        ref_.get_item().data = Hole::Empty(lvl2);
+                        ref_.get_item().data = Hole::Empty(lvl);
```

Comment:
Here, `lvl` is always the value we get from `ref_.get()` in this previous part of the function:
```rs
match ref_.get() {
    Hole::Empty(lvl) =>
```

So, the line substitute the `_ref` data that is `Hole::Empty(lvl)` with `Hole::Empty(lvl)`.
Could you explain what the purpose of this assignment should be?

Reply:
It's probably not correct... I'll let you solve this :3